pulp_ansible/app/tasks/publishing.py

```diff
@@ -2,72 +2,25 @@
 
 from gettext import gettext as _
 
-from pulpcore.plugin.models import (
-    RepositoryVersion,
-    Publication,
-    PublishedArtifact,
-    RemoteArtifact)
-from pulpcore.plugin.tasking import WorkingDirectory
-
-from pulp_ansible.app.models import AnsiblePublisher
+from pulpcore.plugin.models import RepositoryVersion, Publication
 
 
 log = logging.getLogger(__name__)
 
 
-def publish(publisher_pk, repository_version_pk):
+def publish(repository_version_pk):
     """
-    Use provided publisher to create a Publication based on a RepositoryVersion.
+    Create a Publication based on a RepositoryVersion.
 
     Args:
-        publisher_pk (str): Use the publish settings provided by this publisher.
         repository_version_pk (str): Create a publication from this repository version.
     """
-    publisher = AnsiblePublisher.objects.get(pk=publisher_pk)
     repository_version = RepositoryVersion.objects.get(pk=repository_version_pk)
-
-    log.info(
-        _('Publishing: repository=%(repository)s, version=%(version)d, publisher=%(publisher)s'),
-        {
-            'repository': repository_version.repository.name,
-            'version': repository_version.number,
-            'publisher': publisher.name,
-        })
-
-    with WorkingDirectory():
-        with Publication.create(repository_version, publisher) as publication:
-            populate(publication)
+    with Publication.create(repository_version, pass_through=True) as publication:
+        pass
```

Comment:
There's two things this block does for us. One is to set publication as the created_resource on the task and also it handles setting complete on the publication.

Another option would be to copy over this code to do something like:

```python
with transaction.atomic():
  publication = Publication(repository_version=repository_version, pass_through=True, complete=True)
  publication.save()
  resource = CreatedResource(content_object=publication)
  resource.save()
```

Thoughts?